ru/turing/courses/lesson3/Registry.java

```diff
@@ -1,13 +1,111 @@
+/**
+ * Registry class
+ *
+ * @autor Ivanov Andrew
+ * @version 1.0
+ */
+
 package ru.turing.courses.lesson3;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
+
+public class Registry<T extends User> {
+    private final Map<String, T> storage = new HashMap<>();
+
+    /**
+     * Method of adding an item to the registry
+     *
+     * @param entry user for adding
+     */
+    public void add(T entry) {
+        storage.put(entry.getKey(), entry);
+    }
+
+    /**
+     * Method of removing an item by key
+     *
+     * @param key key by which we remove values
+     */
+    public T removeByKey(String key) {
+        return storage.remove(key);
+    }
+
+    /**
+     * Method of deleting items by value
+     *
+     * @param User user with value that all removable users have
+     * @return removed users keys
+     */
+    public List<String> removeByValue(T User) {
```

Comment:
```suggestion
    public List<String> removeByValue(T user) {
```
T -- это тип, после него следует название объекта, а оно с маленькой буквы